pkg/netstate/netstate.go

```diff
@@ -39,10 +41,23 @@ type DB interface {
 	Delete([]byte) error
 }
 
+func validateAuth(xAPIKeyBytes []byte) error {
+	if !auth.ValidateAPIKey(string(xAPIKeyBytes)) {
+		return grpc.Errorf(codes.Unauthenticated, "Invalid API credential")
+	}
+	return nil
+}
+
 // Put formats and hands off a file path to be saved to boltdb
 func (s *Server) Put(ctx context.Context, putReq *pb.PutRequest) (*pb.PutResponse, error) {
 	s.logger.Debug("entering netstate put")
 
+	xAPIKeyBytes := []byte(putReq.XApiKey)
+	if err := validateAuth(xAPIKeyBytes); err != nil {
+		s.logger.Error("unauthorized request")
```

Comment:
Here and in the other requests, you could also log the `err` itself so 57 would be `s.logger.Error("unauthorized request: ", zap.Error(err))`. This isn't required though, just could help in the future!

Reply:
suggested some simplification strategies for this in slack. since this is duplicated in every function, things to make it shorter are good!